server/service/service.go

```diff
@@ -58,13 +58,54 @@ type SecurityArtifacts struct {
 	TLSKeypair *tls.Certificate
 }
 
-// EntityLookup provides a way to resolve chassis and control cards
-// in the EntityManager.
+// EntityLookup is used to resolve the fields of an active control card to a chassis.
+// For fixed form factor devices, the active control card is the chassis itself.
 type EntityLookup struct {
+	// The manufacturer of this control card or chassis.
 	Manufacturer string
+	// The serial number of this control card or chassis.
 	SerialNumber string
-	PartNumber   string
-	IPAddress    string
+	// The hardware model/part number of this control card or chassis.
+	PartNumber string
+	// The reported IP address of the management interface for this control
+	// card or chassis.
+	IPAddress string
+	// Whether this chassis appears to be a modular device.
+	Modular bool
+}
+
+// buildEntityLookup constructs an EntityLookup object from a bootstrap request.
+func buildEntityLookup(ctx context.Context, req *bpb.GetBootstrapDataRequest) (*EntityLookup, error) {
+	peerAddr, err := peerAddressFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	activeControlCardSerial := req.GetControlCardState().GetSerialNumber()
+	var partNumber string
+	var modular bool
+	if len(req.GetChassisDescriptor().GetControlCards()) == 0 {
+		modular = false
+		partNumber = req.GetChassisDescriptor().GetPartNumber()
```

Comment:
would this just be an empty string?

Reply:
Do you mean line 88? (GitHub does a bad job at showing what the comment is anchored to). 

It shouldn't be empty, the proto says that you can either populate the control cards (for modular) or populate chassis descriptor fields (for fixed form factor).

Reply:
I would advise either adding a check that it's not empty if it shouldn't be, or at least adding a comment that it shouldn't be.

Reply:
We check if part number is set for modular or fixed form factor on line 101. So at least one of these places must have it populated.

Reply:
okedoke